### PR TITLE
[tests] Fix reference gas price for sui-core tests

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

- Fixed sui-core tests with RGP of 1

## Test Plan 

- Updated tests 

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-core/src/unit_tests/gas_tests.rs [hunk 1]
```diff
@@ -337,7 +338,7 @@ async fn test_oog_computation_oog_storage() -> SuiResult {
 // - computation ok, OOG for storage, minimal storage ok
 #[tokio::test]
 async fn test_computation_ok_oog_storage_minimal_ok() -> SuiResult {
-    const GAS_PRICE: u64 = 30;
+    const GAS_PRICE: u64 = 1000;
```
Comment (dariorussi) [inline crates/sui-core/src/unit_tests/gas_tests.rs:341]: fwiw I wanted to use values different than 1 or 1000, but it's all good

**Review by dariorussi — APPROVED** [review]
thanks it's getting easier to change the gas value for those tests, that is wonderful!

File: crates/sui-core/src/unit_tests/authority_tests.rs [hunk 2]
```diff
@@ -5228,7 +5229,14 @@ async fn test_for_inc_201_dry_run() {
     builder.publish_immutable(modules, BuiltInFramework::all_package_ids());
     let kind = TransactionKind::programmable(builder.finish());
 
-    let txn_data = TransactionData::new_with_gas_coins(kind, sender, vec![], 50_000_000, 1);
+    let rgp = fullnode.reference_gas_price_for_testing().unwrap();
+    let txn_data = TransactionData::new_with_gas_coins(
+        kind,
+        sender,
+        vec![],
+        TEST_ONLY_GAS_UNIT_FOR_GENERIC * 10,
```
Comment (longbowlu) [inline crates/sui-core/src/unit_tests/authority_tests.rs:5237]:
```suggestion
        TEST_ONLY_GAS_UNIT_FOR_GENERIC * rgp,
```

File: crates/sui-core/src/test_utils.rs [hunk 3]
```diff
@@ -310,7 +310,7 @@ pub fn make_transfer_object_transaction(
         object_ref,
         sender,
         gas_object,
-        gas_price * TEST_ONLY_GAS_UNIT_FOR_TRANSFER,
+        gas_price * TEST_ONLY_GAS_UNIT_FOR_TRANSFER * 10,
```
Comment (longbowlu) [inline crates/sui-core/src/test_utils.rs:313]: is `transfer_object` this expensive?

File: crates/sui-core/src/unit_tests/authority_tests.rs [hunk 4]
```diff
@@ -5532,7 +5554,14 @@ async fn test_publish_not_a_package_dependency() {
     builder.publish_immutable(modules, deps);
     let kind = TransactionKind::programmable(builder.finish());
 
-    let txn_data = TransactionData::new_with_gas_coins(kind, sender, vec![gas_ref], 10000, 1);
```
Comment (longbowlu) [inline crates/sui-core/src/unit_tests/authority_tests.rs:5535]: I think I missed these tests.

File: crates/sui-core/src/unit_tests/gas_tests.rs (hunk 1 above)
Comment (longbowlu) [reply crates/sui-core/src/unit_tests/gas_tests.rs:341]: 1001

File: crates/sui-types/src/sui_system_state/epoch_start_sui_system_state.rs [hunk 5]
```diff
@@ -93,7 +93,7 @@ impl EpochStartSystemStateV1 {
         Self {
             epoch,
             protocol_version: ProtocolVersion::MAX.as_u64(),
-            reference_gas_price: 1,
+            reference_gas_price: 1000,
```
Comment (longbowlu) [inline crates/sui-types/src/sui_system_state/epoch_start_sui_system_state.rs:96]: use the constant? (sth like "DEFAULT_VALIDATOR_GAS_PRICE")

**Review by longbowlu — COMMENTED** [review]
thanks!

File: crates/sui-core/src/test_utils.rs (hunk 3 above)
Comment (tnowacki) [reply crates/sui-core/src/test_utils.rs:313]: I do not know, but it did not pass without this

**Pull request merged**